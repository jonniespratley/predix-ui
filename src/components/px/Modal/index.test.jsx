import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import PxModal from './';

describe('Modal', () => {
  test('should render', () => {
    const onClick = sinon.spy();
    const wrapper = shallow(<PxModal onBackdropClick={onClick}>Content</PxModal>);
    // wrapper.find('.px-overlay').simulate('click');
    //expect(onClick.calledOnce).toEqual(true);
    const tree = renderer.create(<PxModal
      headerText='Modal'
      acceptText='Confirm'
      reject='Cancel'
      />
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
