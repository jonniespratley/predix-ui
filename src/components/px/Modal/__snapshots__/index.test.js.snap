// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render 1`] = `
.c1 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 900;
  background-color: var(--px-modal-overlay-color,rgba(0,0,0,0.5));
  -webkit-transition: background-color 0.2s cubic-bezier(.78,.13,.16,.87);
  transition: background-color 0.2s cubic-bezier(.78,.13,.16,.87);
  visibility: hidden;
  background-color: transparent;
}

.c5 {
  font-family: "GE Inspira Sans";
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  display: inline-block;
  overflow: visible;
  height: var(--px-btn-height,auto);
  min-height: var(--px-btn-min-height,2rem);
  min-width: var(--px-btn-min-width,4.66667em);
  margin: 0;
  border: 1px solid var(--px-btn-border-color,transparent);
  border-radius: 0 !important;
  padding: calc(var(--px-btn-height,2em) / 10) calc(var(--px-btn-height,2em) / 2);
  box-shadow: var(--px-btn-shadow--light,none);
  font: inherit;
  line-height: calc(var(--px-btn-height,2em) - 2px);
  cursor: pointer;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: none;
  white-space: nowrap;
  background-color: var(--px-btn-background,#d8e0e5);
  -webkit-transition: background .4s,border-color .4s,color .4s;
  transition: background .4s,border-color .4s,color .4s;
  -webkit-appearance: button;
  -webkit-font-smoothing: antialiased;
  color: var(--px-btn-color,#2c404c);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: var(--px-btn-primary-color,white);
  border-color: var(--px-btn-primary-border-color,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-primary-background,#364c59);
}

.c5:link,
.c5:visited,
.c5:hover,
.c5:active {
  color: var(--px-btn-color,#2c404c);
}

.c5:hover,
.c5:focus {
  border-color: var(--px-btn-border-color--hover,transparent);
  box-shadow: var(--px-btn-shadow,none);
  background-color: var(--px-btn-background--hover,#a3b5bf);
}

.c5:active {
  outline: none;
  border-color: var(--px-btn-border-color--pressed,transparent);
  box-shadow: none;
  background-color: var(--px-btn-background--pressed,#889aa5);
}

.c5:link,
.c5:visited,
.c5:hover,
.c5:active {
  color: var(--px-btn-primary-color,white);
}

.c5:hover,
.c5:focus {
  border-color: transparent;
  background-color: var(--px-btn-primary-background--hover,#23343f);
}

.c5:active {
  border-color: transparent;
  background-color: var(--px-btn-primary-background--pressed,#121f26);
}

.c0 {
  position: absolute;
  background-color: transparent;
  visibility: hidden;
}

.c2 {
  z-index: 900;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  max-height: calc(100% - 2rem);
  max-width: calc(100% - 2rem);
  overflow: auto;
  white-space: normal;
  word-wrap: break-word;
  background-color: var(--px-modal-background-color,white);
  color: var(--px-modal-text-color,black);
  box-shadow: 0 6px 12px 0 rgba(0,0,0,0.2);
  padding: 1rem;
  position: fixed;
}

.c3 {
  margin-bottom: 1rem;
  margin-top: 0;
  font-size: 1.33333rem;
  line-height: 1;
  font-weight: 400;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-top: 1rem;
  border-top: 1px solid var(--px-modal-border-color,transparent);
}

.c4 button:nth-last-of-type(2) {
  margin-right: 1rem;
}

@media screen and (max-width:44.9375em) {
  .c2 {
    width: 100%;
  }
}

@media screen and (min-width:45em) {
  .c2 {
    min-width: 400px;
  }
}

<div
  className="c0"
>
  <div
    className="px-overlay c1"
    onClick={[Function]}
    onKeyUp={[Function]}
  />
  <div
    className="c2"
    role="region"
  >
    <div>
      <h3
        className="c3"
      >
        Modal
      </h3>
      <div
        className=""
      />
      <div
        className="c4"
      >
        <button
          className="px-button c5"
          disabled={null}
          icon={null}
          id="btnModalPositive"
          onClick={null}
          size={null}
          type={null}
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
